synchronizer/views/auth.py

```diff
@@ -63,6 +62,12 @@ def oauth_callback(provider):
     """
     oauth = OAuthSignIn.get_provider(provider)
     user_info = oauth.callback()
+
+    domains = current_app.config['ALLOWED_REGISTRATION_DOMAINS']
+    if not any(domain in user_info['email'] for domain in domains):

```

Comment:
Not the best solution.

Just an example:
ALLOWED_REGISTRATION_DOMAINS = ['ventor.tech', ]

User tries to login/register with email: ventor.tech.spam@gmail.com